analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/SymbolicContext.cs

```diff
@@ -33,8 +33,9 @@ public class SymbolicContext
         public SymbolicContext(SymbolicValueCounter symbolicValueCounter, IOperationWrapperSonar operation, ProgramState state)
         {
             this.symbolicValueCounter = symbolicValueCounter ?? throw new ArgumentNullException(nameof(symbolicValueCounter));
-            Operation = operation ?? throw new ArgumentNullException(nameof(operation));
             State = state ?? throw new ArgumentNullException(nameof(state));
+
+            Operation = operation; // Operation can be null for the branch nodes.
```

Comment:
Let's restore the order. It's just plain copying values so removing the `??` without the additional empty line was enough. The comment could be used instead of the `??` part or on the field itself.

To illustrate the ordering, the old version can be read as
```
a
b
c
a = a
b = b
c = c
```
while the new one is irregular - harder to read
```
a
b
c
a = a
c = c

b = b
```